### PR TITLE
Add Vesting to Asset Hubs

### PR DESCRIPTION
**Pull request by joepetrowski**
Adds the Vesting pallet to the Asset Hub runtimes, as a few people have requested this feature, and in general the Asset Hubs should provide more balances/asset-related features than the Relay Chain.

### CODE REVIEW

File: system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs [hunk 1]
```diff
@@ -229,6 +231,23 @@ impl pallet_balances::Config for Runtime {
...
+}
+
+impl pallet_vesting::Config for Runtime {
+	type RuntimeEvent = RuntimeEvent;
+	type Currency = Balances;
+	type BlockNumberToBalance = ConvertInto;
+	type MinVestedTransfer = MinVestedTransfer;
+	type WeightInfo = weights::pallet_vesting::WeightInfo<Runtime>;
+	type UnvestedFundsAllowedWithdrawReasons = UnvestedFundsAllowedWithdrawReasons;
+	type BlockNumberProvider = System;
```
Comment (joepetrowski) [inline system-parachains/asset-hubs/asset-hub-kusama/src/lib.rs:247]: TODO: Switch to Relay Chain

**Comment by xlc** [comment]
Why will anyone else other than W3F want to perform vested transfer of DOT? When people requesting vesting, presumably they want to vest transfer assets in pallet-assets, not the relaychain token.

**Comment by joepetrowski** [comment]
> Why will anyone else other than W3F want to perform vested transfer of DOT? When people requesting vesting, presumably they want to vest transfer assets in pallet-assets, not the relaychain token.

For one, the person who requested this was from Talisman, not W3F, for DOT.

Two, it would be great if Parity and W3F used Asset Hub instead of the Relay Chain for vested DOT transfers.

Three, yes I understand the demand for vested transfers of assets. This PR does not prevent that.

**Comment by acatangiu** [comment]
> Why will anyone else other than W3F want to perform vested transfer of DOT? When people requesting vesting, presumably they want to vest transfer assets in pallet-assets, not the relaychain token.

I agree we _also_ need vesting for pallet-assets instances too, not just pallet-balances.

example community request: https://github.com/paritytech/polkadot-sdk/issues/4090

**Review by acatangiu — CHANGES_REQUESTED** [review]
add vesting to pallet-assets instances too

**Comment by joepetrowski** [comment]
> add vesting to pallet-assets instances too

That is the plan, but the Vesting pallet requires something with `LockableCurrency`, which `pallet-assets` does not implement. In order to have vesting for assets, we need:

1. Migration of Vesting pallet from `Currency` to `Fungible`;
2. An implementation of `fungible::freeze` for `pallet-assets` (https://github.com/paritytech/polkadot-sdk/pull/3951)

I don't think we should block the feature of vesting DOT on waiting for these. Asset vesting will probably be another instance of the Vesting pallet anyway, so they have no overlap.

**Review by acatangiu — APPROVED** [review]
adding vesting in two steps: this PR pallet-balances, future PR pallet-assets is fine

**Review by bkchr — APPROVED** [review]

**Pull request merged**